From ed1c88d04a02933f1631fe2366700d5fb08518bf Mon Sep 17 00:00:00 2001
From: Jeremy Compostella <jeremy.compostella@intel.com>
Date: Wed, 8 May 2019 16:35:21 -0700
Subject: [PATCH] usb: fix automatic transport layer fallback

Commit 52965e559cf48302af68b54b6d55ed3e514e07fa broke the transport
layer fallback mecanism.

The usb_start should run EFI_UNSUPPORTED if the no USB protocol are
support (and not EFI_NOT_FOUND).  The transport_start() function
depend on this particular error to decide to fallback on the next
available protocol implementation which is currently TCP if the UEFI
BIOS support this protocol.

Tracked-On: OAM-80257
Signed-off-by: Jeremy Compostella <jeremy.compostella@intel.com>
---
 libefiusb/usb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libefiusb/usb.c b/libefiusb/usb.c
index 78b2eaf..1555af6 100644
--- a/libefiusb/usb.c
+++ b/libefiusb/usb.c
@@ -445,7 +445,7 @@ EFI_STATUS usb_start(UINT8 subclass, UINT8 protocol,
 		polling_based = TRUE;
 		error(L"Self implemented USB device mode protocol running");
 #else
-		return ret;
+		return EFI_UNSUPPORTED;
 #endif // USE_SELF_USB_DEVICE_MODE_PROTOCOL
 	}
 
-- 
2.20.1

